streams/src/test/java/org/apache/kafka/streams/kstream/TimeWindowsTest.java

```diff
@@ -66,38 +66,26 @@ public void windowSizeMustNotBeNegative() {
         TimeWindows.of(ofMillis(-1));
     }
 
-    @Test
+    @Test(expected = IllegalArgumentException.class)
     public void advanceIntervalMustNotBeZero() {
         final TimeWindows windowSpec = TimeWindows.of(ofMillis(ANY_SIZE));
-        try {
-            windowSpec.advanceBy(ofMillis(0));
-            fail("should not accept zero advance parameter");
-        } catch (final IllegalArgumentException e) {
-            // expected
-        }
+        windowSpec.advanceBy(ofMillis(0));
+        fail("should not accept zero advance parameter");
```

Comment:
Why did you change this? If `TimeWindows.of(ofMillis(ANY_SIZE));` would throw `IllegalArgumentException` due to a bug, this test would pass after the change but it should fail.

If you want to improve this tests, you should check the error message of `e` in the catch block, to see if the exception message the the expected one. Similar below.

Reply:
I changed the `try/catch` to `expected` attribute since we are not doing anything after catching the exception. 

This test will fail, if `windowSpec.advanceBy(ofMillis(0));` does not throw `IllegalArgumentException`. I think this is same as previous functionality.

Reply:
That's not the point. We should use try-catch to make sure the right part of the test code throws. If the first line throws, the test **must** fail, but with the change it would pass.